src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_AssignmentOperator.cs

```diff
@@ -301,11 +301,12 @@ private BoundExpression VisitAssignmentOperator(BoundAssignmentOperator node, bo
             if (setMethod is null)
             {
                 var autoProp = (SourcePropertySymbolBase)property.OriginalDefinition;
-                Debug.Assert(autoProp.IsAutoPropertyWithGetAccessor,
+                Debug.Assert(autoProp.SetMethod is null &&
```

Comment:
>autoProp.SetMethod is null && [](http://example.com/codeflow?start=29&length=29)

Is this a duplication of the ```if``` condition above?#

Reply:
Simplified assertion